pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -241,7 +241,9 @@ func newLibvirtDomainManager(connection cli.Connection, virtShareDir, ephemeralD
 		}
 
 		if len(list) == 0 {
-			return nil, nil
+			// We must return an error, otherwise the nil value will be cached
+			// and will make the scraper to fail
```

Comment:
WDYT about removing the comment?
or changing it to something like "in case libvirt is not responsive, `GetDomainStats()` would return an error whenever the cache is not used" to emphasize this is done generally to keep backward compatibility

Reply:
I agree, I just think the comment should be more general and not mention the scraper specifically.

Reply:
changed, wdyt?

Reply:
Thank you! Looks great!